src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -205,20 +214,25 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | "[" ~> cond <~ "]"
     )
 
-  def cqBodyAtom: Parser[Atom] =
-    relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
-      case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  def atom = relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
+    case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  }
+  def modifierAtom = (opt("NOT") ~ "EXIST" | "OUTER") ~ "[" ~ rep1sep(cqBody, ",") ~ "]" ^^ { case (m ~ _ ~ b ~ _) =>
```

Comment:
That makes sense. Simply negating in parser is kind of abusing parser's responsibility, and cause Printer to print different stuff. I think the desugaring step can happen in compilation, as it actually belongs to the compilation process.